### PR TITLE
Support multiple upgrade and downgrade test scenarios

### PR DESCRIPTION
**Pull request by lei-tang**
- Add support of three test scenarios: upgrade-downgrade, upgrade, and downgrade.
- Document how to run different test scenarios with example commands.

### CODE REVIEW

File: upgrade/README.md [hunk 1]
```diff
@@ -1,23 +1,54 @@
...
     cd ROOT-OF-REPO/upgrade
-    export SOURCE_TAG=1.5_latest; export TARGET_TAG=master; export INSTALL_OPTIONS=istioctl; export UPGRADE_TEST_LOCAL=true;
     ```
 
-- Run the upgrade test.
+You may choose three types of test scenarios by configuring
+the TEST_SCENARIO variable. In the following example commands,
+*SOURCE_TAG* specifies the
+version of Istio to be upgraded/downgraded and *TARGET_TAG* specifies the
+version of Istio to which the test will upgrade/downgrade.
```
Comment (carolynhu) [inline upgrade/README.md:22]: nit: `*TARGET_TAG* specifies the version of Istio the test will be upgraded/downgraded to`

File: upgrade/README.md [hunk 2]
```diff
@@ -1,23 +1,54 @@
...
-on your test cases.
+Enter the directory containing the tests.
 
     ```bash
     cd ROOT-OF-REPO/upgrade
-    export SOURCE_TAG=1.5_latest; export TARGET_TAG=master; export INSTALL_OPTIONS=istioctl; export UPGRADE_TEST_LOCAL=true;
     ```
 
-- Run the upgrade test.
+You may choose three types of test scenarios by configuring
```
Comment (carolynhu) [inline upgrade/README.md:18]: nit: list those `three` test scenarios here. I mean at the initial part, we need to declare which three types of test scenarios we support.

File: upgrade/README.md [hunk 3]
```diff
@@ -1,23 +1,54 @@
...
+the TEST_SCENARIO variable. In the following example commands,
+*SOURCE_TAG* specifies the
+version of Istio to be upgraded/downgraded and *TARGET_TAG* specifies the
+version of Istio to which the test will upgrade/downgrade.
+Note that you may need to configure the variables in the commands based
+on your test cases.
+
+* When TEST_SCENARIO is configured as *upgrade-downgrade*,
+during testing, Istio will be upgraded and then downgraded.
```
Comment (carolynhu) [inline upgrade/README.md:27]: nit: `during testing` can be removed for concise. same for others

File: upgrade/README.md (hunk 1 above)
Comment (lei-tang) [reply upgrade/README.md]: Done, thanks!

File: upgrade/README.md (hunk 2 above)
Comment (lei-tang) [reply upgrade/README.md]: Done, thanks!

File: upgrade/README.md (hunk 3 above)
Comment (lei-tang) [reply upgrade/README.md]: Done.

File: upgrade/run_upgrade_test.sh [hunk 4]
```diff
@@ -40,6 +40,14 @@ export FROM_PATH=${FROM_PATH:-"$(mktemp -d from_dir.XXXXXX)"}
 export TO_PATH=${TO_PATH:-"$(mktemp -d to_dir.XXXXXX)"}
 export SOURCE_LINUX_TAR_SUFFIX=${SOURCE_LINUX_TAR_SUFFIX:-"linux-amd64.tar.gz"}
 export TARGET_LINUX_TAR_SUFFIX=${TARGET_LINUX_TAR_SUFFIX:-"linux-amd64.tar.gz"}
+# TEST_SCENARIO has a default value of "upgrade-downgrade", which corresponds to
+# the default test scenario of upgrade followed by downgrade.
+# TEST_SCENARIO can also be configured to "upgrade" or "downgrade", which corresponds to
+# the upgrade-only test scenario and the downgrade-only test scenario, respectively.
+# When the test scenario is "downgrade",SOURCE_HUB and SOURCE_TAG specifies the version
```
Comment (carolynhu) [inline upgrade/run_upgrade_test.sh:47]: nit: `specify`, same for TARGET

File: upgrade/test_crossgrade.sh [hunk 5]
```diff
@@ -127,6 +127,13 @@ if [[ -z "${FROM_HUB}" || -z "${FROM_TAG}" || -z "${FROM_PATH}" || -z "${TO_HUB}
     exit 1
 fi
 
+if [[ "${TEST_SCENARIO}" == "upgrade-downgrade" || "${TEST_SCENARIO}" == "upgrade" || "${TEST_SCENARIO}" == "downgrade" ]];then
+    echo "The test scenario is ${TEST_SCENARIO}."
```
Comment (carolynhu) [inline upgrade/test_crossgrade.sh:131]: nit: either `The current test scenario is ${TEST_SCENARIO}` or `The test scenarios is set as ${TEST_SCENARIO}`

File: upgrade/run_upgrade_test.sh (hunk 4 above)
Comment (lei-tang) [reply upgrade/run_upgrade_test.sh]: Done.

File: upgrade/test_crossgrade.sh (hunk 5 above)
Comment (lei-tang) [reply upgrade/test_crossgrade.sh]: Done.

File: upgrade/test_crossgrade.sh [hunk 6]
```diff
@@ -475,8 +482,10 @@ istioInstallOptions
 waitForIngress
 waitForPodsReady "${ISTIO_NAMESPACE}"
 
-# Make a copy of the "from" sidecar injector ConfigMap so we can restore the sidecar independently later.
-echo_and_run kubectl get ConfigMap -n "${ISTIO_NAMESPACE}" istio-sidecar-injector -o yaml > ${TMP_DIR}/sidecar-injector-configmap.yaml
+if [[ "${TEST_SCENARIO}" == "upgrade-downgrade" ]];then
+  # Make a copy of the "from" sidecar injector ConfigMap so we can restore the sidecar independently later.
```
Comment (carolynhu) [inline upgrade/test_crossgrade.sh:486]: nit: `so that`, sorry I might made too many picky comments.

Comment (lei-tang) [reply upgrade/test_crossgrade.sh]: Done, thank you for your attention to the details and the quality!

**Comment by JimmyCYJ** [comment]
Thanks @lei-tang

**Pull request merged**